docs/testplan/ecn/WRED-and-ECN-statistics-testplan.md

```diff
@@ -0,0 +1,254 @@
+        counterpoll wred-port enable/disable<br>
+
+ 4. on egress port  set   attribute SAI_PORT_ATTR_PKT_TX_ENABLE = false . <br>
+    ( It will build up the traffic in egress queue thereby congestion gets simulated and DUT reacts with WRED/ECN if configured )
+
+ 5. Reset counters using the CLI:<br>
+        sonic-clear queue wredcounters
+
+ 6. Send TCP/UDP traffic switched/routed to the egress interface which is in tx disable state .
+
+ 7. on egress port  set   attribute SAI_PORT_ATTR_PKT_TX_ENABLE = true . <br>
+
+ 8. on supported platform:<br>
+      Verify WRED/ECN counters at queue level using the  new CLI:<br>
+        show queue wredcounters Ethernet&lt;&gt;
+
+      Verify WRED statistics at port level using the enhanced CLI:<br>
+        show interfaces counters detailed Ethernet&lt;&gt;
+
+
+
+
+ Test cases
+ ----------
+ WRED/ECN statistics verification on supported platform
+
+ Terminology:<br>
+    WRED=0/1 refers WRED disabled/enabled on queue <br>
+    ECN=0/1 refers ECN disabled/enabled queue <br>
+    CODE POINT refers to ECN CODE POINT in the IP header <br>
+ | <div style="width:250px">Testcase</div> | <div style="width:300px">Test expectation </div>                                               | <div style="width:500px">Test Steps</div> |
+ | --------------------------------------- | -----------------------------------------------------------------------------------------------| ----------------------------------------- |
+ | WRED=0 ECN=0 CODE POINT=XX              | Tail Drop and hence no WRED/ECN counter increments                                                                 | 1. Query STATE_DB for WRED/ECN support on the platform <br> 2. tx disable egress port  <br>    3. Send TCP/UDP traffic. <br> 4. tx enable egress port <br> 5. Verify WRED/ECN packet counter and Byte counter <br>        |
+ | WRED=1 ECN=0 CODE POINT=XX              | If WRED Algorithm determines pkt is drop eligible then increment WRED counters. ECN counters should not increment . | 1. Query STATE_DB for WRED/ECN support on the platform <br> 2. tx disable egress port  <br>    3. Send TCP/UDP traffic. <br> 4. tx enable egress port <br> 5. Verify WRED/ECN packet counter and Byte counter <br>  |
+ | WRED=1 ECN=1 CODE POINT=00              | If WRED Algorithm determines pkt is drop eligible then increment WRED counters. ECN counters should not increment . | 1. Query STATE_DB for WRED/ECN support on the platform <br> 2. tx disable egress port  <br>    3. Send TCP/UDP traffic. <br> 4. tx enable egress port <br> 5. Verify WRED/ECN packet counter and Byte counter <br> |
+ | WRED=1 ECN=1 CODE POINT=01              | If WRED Algorithm determines pkt is drop eligible then increment ECN counters. WRED counters should not increment . | 1. Query STATE_DB for WRED/ECN support on the platform <br> 2. tx disable  egress port  <br>    3. Send TCP/UDP traffic. <br> 4. tx enable egress port <br> 5. Verify WRED/ECN packet counter and Byte counter <br> |
```

Comment:
When WRED=1 ECN=1 CODE POINT=01,  Can you double-check the Test expectation?
In my opinion,  WRED Algorithm will not drop packet and then mark the packet, so, increment ECN counters and WRED counters should not increment.